src/gflownet/envs/mol_building_env.py

```diff
@@ -333,12 +348,15 @@ def graph_to_mol(self, g: Graph) -> Mol:
             mp.AddBond(e[0], e[1], d.get("type", BondType.SINGLE))
         mp.CommitBatchEdit()
         Chem.SanitizeMol(mp)
-        return mp
+        # Not sure why, but this seems to find errors that SanitizeMol doesn't, and it saves us trouble downstream,
+        # since MolFromSmiles returns None if the SMILES encoding of the molecule is invalid, which seemingly happens
+        # occasionally (although very rarely) even if RDKit is able to create a SMILES string from `mp`.
+        return Chem.MolFromSmiles(Chem.MolToSmiles(mp))
 
     def is_sane(self, g: Graph) -> bool:
         try:
             mol = self.graph_to_mol(g)
-            assert Chem.MolFromSmiles(Chem.MolToSmiles(mol)) is not None
+            assert mol is not None
         except Exception:
             return False
         if mol is None:
```

Comment:
suggestion: is this code still needed? it seems safe to delete since you are already asserting that `mol is not None`

Reply:
Good point!